utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -386,7 +441,8 @@ impl<'a, 'b> SyscallsInvocator<'a, 'b> {
                 }
                 ProcessedSyscallParams::MemoryArrayLength => {
                     let length;
-                    let upper_limit = mem_size.saturating_sub(1) as i32;
+                    let upper_limit =
+                        mem_size.saturating_sub(MemoryLayout::RESERVED_MEMORY_SIZE) as i32;
```

Comment:
did it get it right, that if memory array ptr is chosen to be somewhere like `let random_ptr = mem_size.saturating_sub(MemoryLayout::RESERVED_MEMORY_SIZE).saturating_sub(2)`, then it's enough to have length greater than 2 to rewrite everything (in case we got syscall with `Mut*` like ptr`?

Reply:
it's possible not to pay attention to that if there's only few syscalls which has `Mut*` array ptr params. Can you check that please?

Reply:
No, it works something like this:
```
upper_limit = mem_size - 256

offset = int_in_range(0..=upper_limit)
length = int_in_range(0..=(upper_limit - offset))
```

For example, let's see what happens at the maximum:
```
upper_limit = mem_size - 256

offset = upper_limit
length = (upper_limit - offset) = 0
```

We can point to reserved memory, but with length 0.